test/toy_object_test.rb

```diff
@@ -3,81 +3,77 @@
     end
 
     describe "accessing instance variables" do
       specify "reading and writing with a Symbol name" do
-        call_method(@object, meth_prefix, :instance_variable_set, :@foo, :bar)
-        assert_equal :bar, call_method(@object, meth_prefix, :instance_variable_get, :@foo)
+        @object.instance_variable_set(:@foo, :bar)
+        assert_equal :bar, @object.instance_variable_get(:@foo)
 
-        call_method(@object, meth_prefix, :instance_variable_set, :@foo, :baz)
-        assert_equal :baz, call_method(@object, meth_prefix, :instance_variable_get, :@foo)
+        @object.instance_variable_set(:@foo, :baz)
+        assert_equal :baz, @object.instance_variable_get(:@foo)
       end
 
       specify "reading and writing with a String name" do
-        call_method(@object, meth_prefix, :instance_variable_set, "@foo", :bar)
-        assert_equal :bar, call_method(@object, meth_prefix, :instance_variable_get, "@foo")
+        @object.instance_variable_set("@foo", :bar)
+        assert_equal :bar, @object.instance_variable_get("@foo")
       end
 
       specify "reading with a Symbol and writing with a String name" do
-        call_method(@object, meth_prefix, :instance_variable_set, "@foo", :bar)
-        assert_equal :bar, call_method(@object, meth_prefix, :instance_variable_get, :@foo)
+        @object.instance_variable_set("@foo", :bar)
+        assert_equal :bar, @object.instance_variable_get(:@foo)
       end
 
       specify "reading with a String and writing with a Symbol name" do
-        call_method(@object, meth_prefix, :instance_variable_set, :@foo, :bar)
-        assert_equal :bar, call_method(@object, meth_prefix, :instance_variable_get, "@foo")
+        @object.instance_variable_set(:@foo, :bar)
+        assert_equal :bar, @object.instance_variable_get("@foo")
       end
 
       specify "reading an unset instance variable returns nil" do
-        assert_nil call_method(@object, meth_prefix, :instance_variable_get, :@foo)
+        assert_nil @object.instance_variable_get(:@foo)
       end
     end
 
     describe "#instance_variables" do
       it "returns the names of existing instance variables" do
-        call_method(@object, meth_prefix, :instance_variable_set, :@bar, :foo)
-        call_method(@object, meth_prefix, :instance_variable_set, :@baz, :foo)
+        @object.instance_variable_set(:@bar, :foo)
+        @object.instance_variable_set(:@baz, :foo)
 
-        assert_equal [:@bar, :@baz], call_method(@object, meth_prefix, :instance_variables)
+        assert_equal [:@bar, :@baz], @object.instance_variables
       end
     end
 
     describe "#instance_variable_defined?" do
       it "returns true if instance variable is set" do
-        call_method(@object, meth_prefix, :instance_variable_set, :@bar, :foo)
-        assert call_method(@object, meth_prefix, :instance_variable_defined?, :@bar)
+        @object.instance_variable_set(:@bar, :foo)
+        assert @object.instance_variable_defined?(:@bar)
```

Comment:
I was going to chime in with “we could use `assert_predicate` here now” (cf `assert_kind_of`), but we can’t, because `assert_predicate` doesn’t support predicates which take an argument 🤷🏻‍♂️